lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -690,29 +691,45 @@ LogicalResult LowerAnnotationsPass::solveWiringProblems(ApplyState &state) {
     moduleModifications[sinkModule].connectionMap[index] = sink;
 
     // Record ports that should be added to each module along the LCA path.
-    // Wire using RefType based on the source-- the RefSend will be of this
-    // type, and we cannot connect RefType's of non-identical types. The final
-    // connect of the resolved ref to the sink will handle any differences.
-    RefType tpe = TypeSwitch<Type, RefType>(problem.source.getType())
-                      .Case<FIRRTLBaseType>([](FIRRTLBaseType base) {
-                        return RefType::get(base);
-                      })
-                      .Case<RefType>([](RefType ref) { return ref; });
-    for (auto sourceInst : sources) {
+    RefType refType = TypeSwitch<Type, RefType>(problem.source.getType())
+                          .Case<FIRRTLBaseType>([](FIRRTLBaseType base) {
+                            return RefType::get(base);
+                          })
+                          .Case<RefType>([](RefType ref) { return ref; });
+    StringRef name;
+    for (auto sourceInst : llvm::reverse(sources)) {
       auto mod = cast<FModuleOp>(instanceGraph.getReferencedModule(sourceInst));
+      if (name.empty()) {
+        if (problem.newNameHint.empty())
+          name = state.getNamespace(mod).newName(
+              getFieldName(getFieldRefFromValue(source), /*nameSafe=*/true)
+                  .first +
+              "__bore");
+        else
+          name = state.getNamespace(mod).newName(problem.newNameHint);
+      }
       moduleModifications[mod].portsToAdd.push_back(
-          {index,
-           {StringAttr::get(
-                context, state.getNamespace(mod).newName(problem.newNameHint)),
-            tpe, Direction::Out}});
+          {index, {StringAttr::get(context, name), refType, Direction::Out}});
```

Comment:
Here `name` is used as the port name, but it's generated in the namespace for the previous iteration's module when boring through multiple layers.

How about just do the `name = state.getNamespace(mod).newName(name)` right before adding to `portsToAdd` in each loop?

Neat that even if there's a conflict the names thereafter will include that, so they chain! :)

Reply:
I refactored this to make the change you're proposing. You are correct about the namespace!